### PR TITLE
MM-28882 Add config to Ignore Guests on AD/LDAP Sync

### PR DESCRIPTION
**Pull request by cwarnermm**
https://mattermost.atlassian.net/browse/MM-28882

Updated:
https://docs.mattermost.com/deployment/sso-saml-ldapsync.html

- Under SAML Configure SAML synchronization with AD/LDAP:
replaced redundant step 2 with new step 2 for Ignore Guest configuration setting
added UI context switch details when remaining steps move from SAML 2.0 to AD/LDAP page)

### CODE REVIEW

File: source/deployment/sso-saml-ldapsync.rst [hunk 1]
```diff
@@ -9,13 +9,13 @@ In addition to configuring SAML sign-in, you can optionally configure synchroniz
...
 
- - If you don't want to enable AD/LDAP sign-in, keep **Enable sign-in with AD/LDAP** as ``false``.
+ - If you don't want to enable AD/LDAP sign-in, go to **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and keep **Enable sign-in with AD/LDAP** as ``false``.
 
-4. Set **Syncronization Interval** to specify how often Mattermost synchronizes SAML user accounts with AD/LDAP. The default setting is 60 minutes. If you want to synchronize immediately after disabling an account, use the "AD/LDAP Synchronize Now" button in **System Console > AD/LDAP** in prior versions or **System Console** > **Authentication** > **AD/LDAP** in versions after 5.12.
-5. To test Mattermost can successfully connect to your AD/LDAP server, click the **AD/LDAP Test** button.
+4. To specify how often Mattermost synchronizes SAML user accounts with AD/LDAP, go to **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and set **Synchronization Interval**. The default setting is 60 minutes. If you want to synchronize immediately after disabling an account, click the **AD/LDAP Synchronize Now** button.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-ldapsync.rst:17]:
```suggestion
4. To specify how often Mattermost synchronizes SAML user accounts with AD/LDAP, go to **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and set **Synchronization Interval**. The default setting is 60 minutes. If you want to synchronize immediately after disabling an account, click **AD/LDAP Synchronize Now**.
```

File: source/deployment/sso-saml-ldapsync.rst [hunk 2]
```diff
@@ -9,13 +9,13 @@ In addition to configuring SAML sign-in, you can optionally configure synchroniz
...
+5. To test Mattermost can successfully connect to your AD/LDAP server, go to **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and click the **AD/LDAP Test** button.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-ldapsync.rst:18]:
```suggestion
5. To confirm that Mattermost can successfully connect to your AD/LDAP server, go to **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and select **AD/LDAP Test**.
```

**Comment by justinegeffen** [comment]
I've made two minor rewording suggestions but both based on preexisting content that I just noticed now. :)

**Review by justinegeffen — APPROVED** [review]

**Comment by cwarnermm** [comment]
Thanks!

File: source/deployment/sso-saml-ldapsync.rst [hunk 3]
```diff
@@ -9,13 +9,13 @@ In addition to configuring SAML sign-in, you can optionally configure synchroniz
 To configure SAML synchronization with AD/LDAP:
 
 1. Go to **System Console > Authentication > SAML 2.0** (or **System Console > SAML** in versions prior to 5.12) and set **Enable Synchronizing SAML Accounts With AD/LDAP** to ``true``.
-2. Go to  **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and set **Enable Synchronization with AD/LDAP** to ``true``.
+2. Go to  **System Console > Authentication > SAML 2.0** and set **Ignore Guest Users when Synchronizing with AD/LDAP** to ``true`` or ``false``.
```
Comment (sbishel) [inline source/deployment/sso-saml-ldapsync.rst:12]: The original step 2 is still valid. It should now be step 3.

**Review by sbishel — CHANGES_REQUESTED** [review]

File: source/deployment/sso-saml-ldapsync.rst (hunk 3 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-ldapsync.rst:12]: Very good catch, Scott. Thank you for the oversight!

Comment (cwarnermm) [reply source/deployment/sso-saml-ldapsync.rst]: Should be resolved with my latest commit. Also updated the ignore step, given its default value.

**Review by sbishel — APPROVED** [review]

**Pull request merged**